### PR TITLE
HOSTEDCP-967: Disable v1alpha1 and conversion webhook by default

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Disables `v1alpha1` and the conversion webhook by default when installing hypershift.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-967](https://issues.redhat.com/browse/HOSTEDCP-967)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/install/install.go [hunk 1]
```diff
@@ -240,17 +239,61 @@ func apply(ctx context.Context, objects []crclient.Object) error {
...
+		if err := client.Patch(ctx, object, crclient.RawPatch(types.ApplyPatchType, objectBytes.Bytes()), crclient.ForceOwnership, crclient.FieldOwner("hypershift")); err != nil {
+			errs = append(errs, err)
+		}
+		fmt.Printf("applied %s %s/%s\n", object.GetObjectKind().GroupVersionKind().Kind, object.GetNamespace(), object.GetName())
 	}
 
 	return errors.NewAggregate(errs)
 }
 
+func patchManagedFields(object crclient.Object, client crclient.Client, ctx context.Context) error {
```
Comment (csrwng) [inline cmd/install/install.go:262]: nit: s/patchManagedFields/patchHyperShiftCRDManagedFields/

**Review by csrwng — COMMENTED** [review]
Just a nit, lgtm

**Comment by enxebre** [comment]
Very cool!
/lgtm

**Comment by sjenning** [comment]
being reverted because of failure in CI https://github.com/openshift/hypershift/pull/2627

4.13 mgmt cluster fails to install with CNO unable to set HCP

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn/1663758898563125248/artifacts/e2e-aws-ovn/dump-management-cluster/artifacts/namespaces/clusters-5348ac4a5940f68ab232-mgmt/core/pods/logs/cluster-network-operator-6bc47d8d57-k5st7-cluster-network-operator.log

```
Failed to set HostedControlPlane: conversion webhook for hypershift.openshift.io/v1beta1, Kind=HostedControlPlane failed: Post "https://operator.hypershift.svc:443/convert?timeout=30s": dial tcp 10.128.4.128:9443: connect: connection refused
```

**Pull request merged**